### PR TITLE
Update readme with updates to the library.

### PR DESCRIPTION
**Pull request by Amplifiyer**
Update readme.md with updates to the library.
 * Add API and GraphQL category as supported.
 * Update code sample based on the refactor changes.
 * Add note for upcoming breaking changes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ We are iterating and looking for feedback and collaboration, so please [**let us
  - [Getting Started Guide](https://docs.amplify.aws/start/getting-started/setup/q/integration/flutter)
  - [Example Application](https://github.com/aws-amplify/amplify-flutter/tree/master/example)
  - [Roadmap/Provide Feedback](https://github.com/aws-amplify/amplify-flutter/issues/5)
+ 
+ ⚠️ **For upcoming breaking changes in 0.0.1-dev.7 please refer this [issue](https://github.com/aws-amplify/amplify-flutter/issues/274) for migration details.**
```
Comment (haverchuck) [inline README.md:17]: Good call

**Review by haverchuck — APPROVED** [review]

**Review by Ashish-Nanda — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jamesonwilliams) [inline README.md:17]: Should it be 0.0.2?

Comment (Amplifiyer) [reply README.md:17]: Makes sense. Will update to 0.0.2

Comment (Amplifiyer) [reply README.md]: Apparently 0.0.2 is not helping, ^0.0.1-dev.6 is still bringing in 0.0.2-dev.1

**Pull request merged**